### PR TITLE
LRQA-57223 | master

### PR DESCRIPTION
**Pull request by SylviaLuan**
https://liferay.atlassian.net/browse/LRQA-57223
https://liferay.atlassian.net/browse/LPS-195179

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/CollectionsWithStaging.testcase [hunk 1]
```diff
@@ -129,4 +129,229 @@ definition {
...
+
+			Segmentation.addProperties(propertyOption = "URL");
+
+			Segmentation.fillTitle(titleField = "ADMIN");
+
+			Segmentation.changeCriterionInput(criterionInputOption = "Contains");
+
+			Segmentation.fillProperty(criterionValue = "admin");
+
+			Button.clickSave();
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/CollectionsWithStaging.testcase:205]: Use json macro
For example
```
			JSONSegmentsentry.addSegment(
				conditionType = "Organization",
				fieldName = "Region",
				groupName = "Test Site Name",
				operator = "equals",
				segmentName = "Org with Region",
				text = "utah");
```

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/CollectionsWithStaging.testcase [hunk 2]
```diff
@@ -129,4 +129,229 @@ definition {
...
+		}
+
+		task ("Add new Manual Selection and select Content A") {
+			AssetListsAdmin.openAssetListsAdmin(siteURLKey = "testsite-staging");
+
+			AssetListsAdmin.addManualSelection(assetListTitle = "DEMO");
+
+			AssetListsAdmin.openAssetListsAdmin(siteURLKey = "testsite-staging");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/CollectionsWithStaging.testcase:213]: No need

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/CollectionsWithStaging.testcase [hunk 3]
```diff
@@ -129,4 +129,229 @@ definition {
...
+
+			AssetListsAdmin.selectAsset(
+				assetListName = "DEMO",
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/CollectionsWithStaging.testcase:216]: Remove this parameter

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/CollectionsWithStaging.testcase [hunk 4]
```diff
@@ -129,4 +129,229 @@ definition {
...
+				locator1 = "AssetLists#VARIATION_INDEX",
+				value1 = "Anyone");
+		}
+
+		task ("Add Asset Publisher to pages, select Collection DEMO") {
+			Navigator.gotoStagedSitePage(
+				pageName = "home",
+				siteName = "testsite");
+
+			Portlet.addPG(portletName = "Asset Publisher");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/CollectionsWithStaging.testcase:253]: Use json macro

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/CollectionsWithStaging.testcase [hunk 5]
```diff
@@ -129,4 +129,229 @@ definition {
...
+
+			AssetPublisherPortlet.selectAssetList(assetListName = "DEMO");
+
+			Navigator._gotoPage(pageName = "admin");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/CollectionsWithStaging.testcase:257]: Navigator.gotoStagedSitePage(
				pageName = "home",
				siteName = "testsite");

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/CollectionsWithStaging.testcase [hunk 6]
```diff
@@ -129,4 +129,229 @@ definition {
...
+
+			Portlet.addPG(portletName = "Asset Publisher");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/CollectionsWithStaging.testcase:259]: Use json macro

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/CollectionsWithStaging.testcase [hunk 7]
```diff
@@ -129,4 +129,229 @@ definition {
...
+
+			JSONWebcontent.addWebContent(
+				content = "Content B",
+				groupName = "testsite",
+				title = "Content B");
+
+			JSONWebcontent.addWebContent(
+				content = "Content C",
+				groupName = "testsite",
+				title = "Content C");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/CollectionsWithStaging.testcase:161]: Use for loop

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/CollectionsWithStaging.testcase [hunk 8]
```diff
@@ -129,4 +129,229 @@ definition {
...
+		task ("Add site with two pages") {
+			HeadlessSite.addSite(siteName = "testsite");
+
+			JSONLayout.addPublicLayout(
+				groupName = "testsite",
+				layoutName = "home");
+
+			JSONLayout.addPublicLayout(
+				groupName = "testsite",
+				layoutName = "admin");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/CollectionsWithStaging.testcase:144]: Use for loop

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/CollectionsWithStaging.testcase [hunk 9]
```diff
@@ -129,4 +129,229 @@ definition {
...
+		}
+
+		task ("Setup a local staging, with Segments staged") {
+			Staging.openStagingAdmin(siteURLKey = "testsite");
+
+			Staging.activateStagingCP(siteName = "testsite");
+
+			Navigator.gotoStagedSitePage(
+				pageName = "home",
+				siteName = "testsite");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/CollectionsWithStaging.testcase:171]: No need

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/CollectionsWithStaging.testcase [hunk 10]
```diff
@@ -129,4 +129,229 @@ definition {
...
+
+			Navigator._gotoPage(pageName = "admin");
+
+			Portlet.addPG(portletName = "Asset Publisher");
+
+			AssetPublisherPortlet.selectAssetList(assetListName = "DEMO");
+		}
+
+		task ("Access pages with the following URLs on the Staging site") {
+			Navigator.openWithAppendToBaseURL(urlAppend = "web/testsite-staging");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/CollectionsWithStaging.testcase:265]: Navigator.openStagingSiteURL

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/CollectionsWithStaging.testcase [hunk 11]
```diff
@@ -129,4 +129,229 @@ definition {
...
+				assetContent = "Content B",
+				assetTitle = "Content B");
+		}
+
+		task ("Access pages with the following URLs on the Living site") {
+			Staging.gotoPublishToLive();
+
+			Staging.publishToLive();
+
+			Navigator.openWithAppendToBaseURL(urlAppend = "web/testsite ");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/CollectionsWithStaging.testcase:313]: Navigator.openSiteURL

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/CollectionsWithStaging.testcase [hunk 12]
```diff
@@ -129,4 +129,229 @@ definition {
...
+
+			AssetPublisherPortlet.viewAssetNotPresentPG(
+				assetContent = "Content B",
+				assetTitle = "Content B");
+
+			AssetPublisherPortlet.viewAssetNotPresentPG(
+				assetContent = "Content C",
+				assetTitle = "Content C");
+
+			Navigator.openWithAppendToBaseURL(urlAppend = "web/testsite/home");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/CollectionsWithStaging.testcase:327]: Navigator.openSitePage

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/CollectionsWithStaging.testcase [hunk 13]
```diff
@@ -129,4 +129,229 @@ definition {
...
+
+			AssetPublisherPortlet.viewAssetNotPresentPG(
+				assetContent = "Content A",
+				assetTitle = "Content A");
+
+			AssetPublisherPortlet.viewAssetNotPresentPG(
+				assetContent = "Content C",
+				assetTitle = "Content C");
+
+			Navigator.openWithAppendToBaseURL(urlAppend = "web/testsite/admin");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/CollectionsWithStaging.testcase:341]: Navigator.openSitePage

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/CollectionsWithStaging.testcase [hunk 14]
```diff
@@ -129,4 +129,229 @@ definition {
 		}
 	}
 
+	@description = "This is a use case for LPS-109102."
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/CollectionsWithStaging.testcase:132]: Missing description

**Review by Tim-Cao — CHANGES_REQUESTED** [review]
Hi @SylviaLuan, could you modify according to comments? Thanks

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**